corehq/apps/export/models/new.py

```diff
@@ -548,10 +548,11 @@ def _get_sub_documents_helper(path, row_docs):
             doc = row_doc.doc
             row_index = row_doc.row
 
-            try:
+            if isinstance(doc, dict):
                 next_doc = doc.get(path[0].name, {})
-            except AttributeError:
-                raise PossibleHeterogenousDataException(doc, path[0].name)
```

Comment:
This is the only place this exception is raised, so you could also drop the exception class and the code which catches it (the soft_assert).

Reply:
Good call. I'll move the soft assert here.